File: typescript/5kyu/SimplePigLatin.ts
/* 
Move the first letter of each word to the end of it, then add "ay" to the end of the word. 
Leave punctuation marks untouched.
*/

export const pigIt = (a : string) : string =>  {
  // split the received string by each word
  let words: string[] = a.split(" ");

  // return the mapped words array
  return words.map((word: string) => {
    // if the current word is alphabetical characters
    // then return the pig latin word
    // else return the word
    return word.match(/[A-z]/i)
      ? `${word.substr(1)}${word.substr(0, 1)}ay` 
      : word
    // join the mapped array of words together by spaces
  }).join(" ");
}

File: c++/5kyu/SimplePigLatin.cpp
/*
Move the first letter of each word to the end of it, then add "ay" 
to the end of the word. Leave punctuation marks untouched.
*/

#include <sstream>
#include <iostream>

std::string pig_it(std::string str)
{
  std::istringstream iss(str);
  std::string s;
  std::vector<std::string> words;
  std::string pigLatin = "";
  std::string fc  = "";
  std::string word = "";
  std::string newStr = "";
  
  while (getline(iss, s, ' ')) {
    words.push_back(s);
  }
  
  int sz = words.size();
  
  for (int i = 0; i < sz; i++) {
    int len = words[i].size();
    fc = words[i].substr(0, 1);
    
    if (fc == "!" || fc == "." || fc == "?" || fc == ",") {
      newStr = fc;
    } else {
      word = words[i].substr(1, len);
      newStr = word + fc + "ay";
    }
    
    pigLatin += newStr;
    
    if (i != sz-1) {
      pigLatin += " ";
    }
  }
  
  return pigLatin;
}
